spec/controllers/posts_controller_spec.rb

```diff
@@ -11,30 +12,33 @@
         it { is_expected.to have_http_status "200" }
     end
 
-    describe "#new" do
-        subject{post :create, :params => { :post => { post_content: "Checking if I can add a post" }, :format => :json}}
+    describe "POST #create" do
+        let(:create_new){post :create, :params => { :post => { post_content: "Checking if I can add a post" }, :format => :json}}
         before do
             @current_user = FactoryBot.create(:user)
             Current.user = @current_user
             sign_in @current_user
         end
-        it { expect{subject}.to change(Post, :count).by(1) }
+        it { expect{create_new}.to change(Post, :count).by(1) }
     end
 
-    describe "#edit" do
+    describe "PATCH #update" do
+        let(:random_post) {Post.create(post_content: "Change me")}
+        let(:update_action) {patch :update, params: { id: random_post.id,
+            random_post => { post_content: 'I can edit'} }; random_post.reload}
```

Comment:
PostControllerのParamsに対応するものなので、
```
def post_params
      params.require(:post).permit(:post_content, :id, :username)
    end
```
paramsは` {id: random_post.id, post: {post_content: 'I can edit'}}` が正しいと思います。

```suggestion
        let(:update_action) {patch :update, params: { id: random_post.id,
           post: {post_content: 'I can edit'} }; random_post.reload}
```